src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+          }
+        }
+
+        // NB: only create the cigar from phased variants.
+        if (isCallVariantPhased || isTruthVariantPhased) {
+          contextsToCigar(truthPhasedVariant, callPhasedVariant)
+        }
+        else Seq.empty
+      }.toList match { // remove leading `*End`
+        case Nil            => Nil
+        case BothEnd :: xs  => xs
+        case CallEnd :: xs  => xs
+        case TruthEnd :: xs => xs
+        case xs             => xs // unreachable?
+     }
+
+    // Update the various rates...
+    metric.finalizeMetric()
+
+    Seq(BothEnd) ++ cigar ++ Seq(BothEnd)
+  }
+
+ /** Gets all the indices in the cigar where there is a short switch error.
+  *
+  * Short switches are sites where the alleles are misphased (flipped), and both upstream and downstream, either the
+  * haplotype block ends, or the next variant has the same phase.
+  */
+  def toShortSwitchErrorIndices(cigar: Cigar): Seq[Int] = {
+    // check if we can find a match or the end of a block at the given idx.  If we cannot, move to the next idx.
+    @tailrec
+    def checkNext(cigar: Seq[CigarType], idx: Int, by: Int): Boolean = {
+      if (idx < 0 || cigar.length <= idx) return true // no more
+      cigar(idx) match {
+        case Match | CallEnd | BothEnd | TruthEnd => true
+        case Mismatch     => false
+        case Call | Truth => checkNext(cigar=cigar, idx=idx+by, by=by)
+      }
+    }
+    // find cigars that have a mismatch at the index, but then anchored on either side
+    cigar.indices.filter { i => cigar(i) == Mismatch && checkNext(cigar, i-1, -1) && checkNext(cigar, i+1, 1) }.toList
+  }
+
+  /** Gets the number of long switch errors and the number of sites examined.
+    *
+    * The number of sites examined are places where there are both a truth and called variant.
+    *
+    * Long switches are a stretch of two or more consecutive sites where the alleles are misphased (flipped), and
+    * both upstream and downstream, either the haplotype block ends, or the next variant has the same phase.
+    */
+  def toLongSwitchErrorsAndSites(cigar: Cigar): (Int, Int) = {
+    // check if we can find a match or the end of a block at the given idx.  If we cannot, move to the next idx.
+    @tailrec
+    def checkNext(cigar: Seq[CigarType], idx: Int, by: Int, stretchLength: Int = 0): Int = {
+      if (idx < 0 || cigar.length <= idx) return stretchLength
+      cigar(idx) match {
+        case Match | CallEnd | BothEnd | TruthEnd => stretchLength
+        case Mismatch     => checkNext(cigar=cigar, idx=idx+by, by=by, stretchLength=stretchLength+1)
+        case Call | Truth => checkNext(cigar=cigar, idx=idx+by, by=by, stretchLength=stretchLength)
+      }
+    }
+
+    // Gets all the indices in the cigar where there is a long switch error. Currently returns the index for all cigar
+    // mismatches that contribute to a long switch error, and tries to maximize such a stretch.
+    val longSwitchErrorIndices = cigar.indices
```

Comment:
You could also, I think, implement this with a [better]buffered iterator over the Cigar, something like:

```scala
val iter = cigar.iterator.bufferBetter
var count = 0
while (iter.hasNext) {
  if (iter.head == MisMatch) {
    val len = iter.takeWhile(c => c == Mismatch || c == Call || c == Truth).filter(_ == Mismatch).length
    if (len > 0) count += 1
  }
  iter.dropWhile(_ != Mismatch)
}
```